python/python_fastapi_server/routers/edr_utils.py

```diff
@@ -0,0 +1,808 @@
+    )
+
+    #   crs_details=[crs_object],
+    position_variables = Variables(
+        query_type="position",
+        default_output_format="CoverageJSON",
+        output_formats=["CoverageJSON"],
+    )
+    position_link = EDRQueryLink(
+        href=f"{base_url}/position",
+        rel="data",
+        hreflang="en",
+        title="Position query",
+        variables=position_variables,
+    )
+    cube_variables = Variables(
+        query_type="position",
+        default_output_format="CoverageJSON",
+        output_formats=["CoverageJSON"],
+    )
+    cube_link = EDRQueryLink(
+        href=f"{base_url}/cube",
+        rel="data",
+        hreflang="en",
+        title="Cube query",
+        variables=cube_variables,
+    )
+
+    locations_variables = Variables(
+        query_type="locations",
+        default_output_format="GeoJSON",
+        output_formats=["GeoJSON"],
+    )
+    locations_link = EDRQueryLink(
+        href=f"{base_url}/locations",
+        rel="data",
+        hreflang="en",
+        title="Locations query",
+        variables=locations_variables,
+    )
+
+    instances_link = None
+    if ref_times and len(ref_times) > 0:
+        instances_variables = Variables(
+            query_type="instances",
+            #  crs_details=[crs_object],
+            default_output_format="CoverageJSON",
+            output_formats=["CoverageJSON", "GeoJSON"],
+        )
+        instances_link = EDRQueryLink(
+            href=f"{base_url}/instances",
+            rel="collection",
+            hreflang="en",
+            title="Instances query",
+            variables=instances_variables,
+        )
+        data_queries = DataQueries(
+            position=EDRQuery(link=position_link),
+            cube=EDRQuery(link=cube_link),
+            instances=EDRQuery(link=instances_link),
+            locations=EDRQuery(link=locations_link),
+        )
+    else:
+        data_queries = DataQueries(
+            position=EDRQuery(link=position_link),
+            cube=EDRQuery(link=cube_link),
+            locations=EDRQuery(link=locations_link),
+        )
+
+    parameter_names = get_params_for_collection(
+        edr_collection=edr_collection, wmslayers=wmslayers
+    )
+
+    crs = ["EPSG:4326"]
+
+    output_formats = ["CoverageJSON"]
+    if instance is None:
+        collection = Collection(
+            links=links,
+            id=edr_collection,
+            extent=extent,
+            data_queries=data_queries,
+            parameter_names=parameter_names,
+            crs=crs,
+            output_formats=output_formats,
+        )
+    else:
+        collection = Instance(
+            links=links,
+            id=instance,
+            extent=extent,
+            data_queries=data_queries,
+            parameter_names=parameter_names,
+            crs=crs,
+            output_formats=output_formats,
+        )
+
+    return collection, ttl
+
+
+def parse_period_string(period: str):
```

Comment:
Can you provide a small descriptive docstring for this function?

Reply:
Done